### ISSUE TITLE
System.Native doesn't support Out Of Tree builds or multiple targets

### ISSUE THREAD

**Issue by kumpera** [mono/mono#10197]
This is an usability issue to anyone working with multiple targets in the same tree as it fails to build once the host changes.

We should either support *not* building it with a configure flags (and use it in the SDKs) or fix it to properly integrate with autotools OOT builds.

**Comment by alexischr** [comment]
This should be fixed with https://github.com/mono/mono/pull/10228,  correct?

**Comment by kumpera** [comment]
No, #10228 just ensures the submodules are checked out. System.Native still built in the bowels of the corefx checkout.
